geomesa-utils-parent/geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/iterators/PlaybackDatastoreIterator.scala

```diff
@@ -0,0 +1,180 @@
+/***********************************************************************
+ * Copyright (c) 2013-2024 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.utils.iterators
+
+import com.typesafe.scalalogging.StrictLogging
+import org.geotools.api.data.{DataStore, Query, Transaction}
+import org.geotools.api.feature.simple.SimpleFeature
+import org.geotools.api.filter.Filter
+import org.geotools.api.filter.sort.SortOrder
+import org.geotools.factory.CommonFactoryFinder
+import org.geotools.feature.simple.SimpleFeatureImpl
+import org.geotools.temporal.`object`.{DefaultInstant, DefaultPeriod, DefaultPosition}
+import org.geotools.util.factory.Hints
+import org.locationtech.geomesa.utils.collection.{CloseableIterator, SelfClosingIterator}
+
+import java.util.Date
+import java.util.concurrent.{Executors, LinkedBlockingQueue}
+import scala.concurrent.duration.Duration
+import scala.util.control.NonFatal
+
+/**
+ * Query over a time frame and return the features in sorted order, delayed based on the date of each feature
+ * to simulate the original ingestion stream
+ *
+ * @param ds data store
+ * @param typeName simple feature type name
+ * @param interval interval to query
+ * @param dtg date attribute to sort by
+ * @param filter additional filter predicate, if any
+ * @param transforms query transforms, if any
+ * @param window length of a single query window, used to chunk up the total features
+ * @param rate multiplier for the rate of returning features, applied to the original delay between features
+ * @param live project dates to current time
+ * @param readAhead size of the read-ahead queue used for holding features before returning them
+ */
+class PlaybackDatastoreIterator(
```

Comment:
to minimize changes to the API, and to keep the diff smaller, could we keep PlaybackIterator the same, and then add a new class for the new iterator? i'm not thinking of any great names for the new class, maybe TimeDelayIterator?